### PR TITLE
Fix support search query reset

### PR DESCRIPTION
**Pull request by getdave**
Alterative approach to https://github.com/Automattic/wp-calypso/pull/43323.

Please see that PR for more details and full testing instructions.

-------------

This PR fixes an issue in the support search for both:

- the "Get help" card on "My Home"
- the blue "?" FAB inline help

The problem is best summarised in [the Issue](https://github.com/Automattic/wp-calypso/issues/43105):

> When interacting with the new support search card on Calypso My Home, resetting the search query to empty causes the "no results" state to appear in the UI.

Fixes https://github.com/Automattic/wp-calypso/issues/43105

### CODE REVIEW

File: client/state/inline-help/reducer.js [hunk 1]
```diff
@@ -70,6 +72,16 @@ export const search = withoutPersistence(
 		action
 	) => {
 		switch ( action.type ) {
+			case INLINE_HELP_SEARCH_RESET:
+				return {
+					searchQuery: '',
+					items: {
+						...state.items,
+						'': action.searchResults,
```
Comment (retrofox) [inline client/state/inline-help/reducer.js:80]: :interesting:

Comment (getdave) [reply client/state/inline-help/reducer.js:80]: Better ideas welcome 👍 

This is how the state is stored. 

Screen Shot 2020-06-18 at 15 00 16

**Comment by getdave** [comment]
FYI I don't believe these Desktop tests are errors relating to these changes in this PR.

File: client/blocks/inline-help/inline-help-search-card.jsx [hunk 2]
```diff
@@ -71,15 +71,14 @@ class InlineHelpSearchCard extends Component {
...
-		if ( ! query || ! query.length ) {
-			return debug( 'empty query. Skip recording tracks-event.' );
-		}
 
-		debug( 'search query received: ', searchQuery );
-		this.props.recordTracksEvent( 'calypso_inlinehelp_search', {
-			search_query: searchQuery,
-			location: 'inline-help-popover',
-		} );
+		if ( query?.length ) {
```
Comment (retrofox) [inline client/blocks/inline-help/inline-help-search-card.jsx:75]: 💯

**Pull request merged**